Assets/MixedRealityToolkit/Providers/Hands/HandRay/AltHandRay.cs

```diff
@@ -0,0 +1,164 @@
+﻿using Microsoft.MixedReality.Toolkit.Utilities;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Alternate hand ray implementation by prvncher (Eric Provencher)
+    ///
+    /// What does this do?
+    /// - Rather than computing hand ray direction by estimating shoulder position by offsetting a point using the head rotation,
+    /// we estimate a torso direction, using head and hand positions, and use that to estimate the shoulder offset.
+    /// 
+    /// Why use this class instead of the original hand ray?
+    /// - More resilient to head rotating while working with menu.
+    /// - Overall more stable, with a more comfortable angle.
+    ///
+    /// How to use this class?
+    /// - Right now, there is no mechanism in place to introduce an alternate hand ray implementation.
+    /// - For now, to get this working, simply replace the class HandRay, by AltHandRay in BaseHand.cs.
+    /// </summary>
+    public class AltHandRay : IHandRay
```

Comment:
nit: I love your addition. Is there a better name you can consider then AltHandRay? It's just a bit along the line of HandRay1, HandRay2, etc. 

I mean HandRay should probably be called BaseHandRay/DefaultHandRay.

I am terrible with names so can't really help with a suggestion. TorsoBasedHandRay? lol

Reply:
To be honest, I'd kind of like to rip out the torso simulation to build a kind of proper IK system, and base this on that.

I'll try and think of a name. Good suggestions :)

Reply:
Ive settled on TorsoDirectedHandRay for now